### ISSUE TITLE
Multiple custom priors not supported for BayesFactor

### ISSUE THREAD

**Issue by mattansb** [easystats/insight#223]
Started looking at #222, came across this - I do love to complicate things 😇

``` r
library(BayesFactor)
iris$ID <- factor(rep(1:30, each = 5))
res <- lmBF(Sepal.Length ~ Species * Sepal.Width + Petal.Width + ID + Sepal.Width:ID, data = iris,
            rscaleCont = c(Sepal.Width = 0.123, Petal.Width = 0.321),
            whichRandom = c("ID","Sepal.Width:ID"))

insight::get_priors(res)
#> Error in data.frame(Parameter = names(prior), Distribution = "cauchy", : arguments imply differing number of rows: 3, 1, 4
```

<sup>Created on 2020-09-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
``` r
library(BayesFactor)
#> Loading required package: coda
#> Loading required package: Matrix
#> ************
#> Welcome to BayesFactor 0.9.12-4.2. If you have questions, please contact Richard Morey (richarddmorey@gmail.com).
#> 
#> Type BFManual() to open the manual.
#> ************
iris$ID <- factor(rep(1:30, each = 5))
res <- lmBF(Sepal.Length ~ Species * Sepal.Width + Petal.Width + ID + Sepal.Width:ID, data = iris,
            rscaleCont = c(Sepal.Width = 0.123, Petal.Width = 0.321),
            whichRandom = c("ID","Sepal.Width:ID"))
#> Warning in rscale[rscaleTypes == "continuous"] <- rscaleCont: number of items to
#> replace is not a multiple of replacement length
#> Warning in doNwaySampling(method, y, X, rscale, iterations, gMap, incCont, :
#> Some NAs were removed from sampling results: 10000 in total.

utils::tail(res@numerator, 1)[[1]]@prior[[1]]
#> $fixed
#> [1] 0.5
#> 
#> $random
#> [1] 1
#> 
#> $continuous
#> Sepal.Width Petal.Width 
#>       0.123       0.321 
#> 
#> $effects
#> NULL
```

What would be the expected output?

Location = 0 for all?
And what would be the names of the priors, the names of the list-elements?

**Comment by mattansb** [comment]
> Location = 0 for all?

Yes

> And what would be the names of the priors, the names of the list-elements?

That depends on what we decide to do with #222 - if that is resolved, than the custom effect priors would be incorporated into that as the correct values.

<details>
<summary><i>Rant...</i></summary>
To be honest, we're giving `BayesFactor` more support than `BayesFactor` is :/ isn't this a little over kill? That package hasn't been updated in over 2 years, most users of it use it in JASP/JAMOVI or don't really care about the various parameters / priors/ etc....  
</details>

**Comment by strengejacke** [comment]
> To be honest, we're giving `BayesFactor` more support than `BayesFactor` is

As far as I remember, all bayesfactor-related functions in _bayestestR_ were implemented by you! :-D
According to the names... since it looks somewhat stupid to have a mix of "fixed" and "Sepal.Width", I'd say we use "fixed", "random" and "continuous" as names...

**Comment by mattansb** [comment]
> As far as I remember, all bayesfactor-related functions in bayestestR were implemented by you! :-D

I would like to see the `blame`s to back that up such a claim! 😅 I was only enabling @IndrajeetPatil - truly my fault 😝

> According to the names... since it looks somewhat stupid to have a mix of "fixed" and "Sepal.Width", I'd say we use "fixed", "random" and "continuous" as names...

But if #222 is fixed according to https://github.com/easystats/insight/issues/222#issuecomment-692643421, than "fixed", "random" and "continuous" won't appear as names any more, so this wont "mix".

**Comment by strengejacke** [comment]
I just looked at the function, I think, no model (not even rstanarm or brms) return a `Component` column - so I would indeed use `fixed` , `random` and `Sepal.Width` and `Petal.Width`, so we can match the right BFs to the parameters later...

**Comment by strengejacke** [comment]
``` r
library(insight)
library(BayesFactor)
#> Loading required package: coda
#> Loading required package: Matrix
#> ************
#> Welcome to BayesFactor 0.9.12-4.2. If you have questions, please contact Richard Morey (richarddmorey@gmail.com).
#> 
#> Type BFManual() to open the manual.
#> ************
iris$ID <- factor(rep(1:30, each = 5))
res <- lmBF(Sepal.Length ~ Species * Sepal.Width + Petal.Width + ID + Sepal.Width:ID, data = iris,
            rscaleCont = c(Sepal.Width = 0.123, Petal.Width = 0.321),
            whichRandom = c("ID","Sepal.Width:ID"))
#> Warning in rscale[rscaleTypes == "continuous"] <- rscaleCont: number of items to
#> replace is not a multiple of replacement length
#> Warning in doNwaySampling(method, y, X, rscale, iterations, gMap, incCont, :
#> Some NAs were removed from sampling results: 10000 in total.

get_priors(res)
#>     Parameter Distribution Location Scale
#> 1       fixed       cauchy        0 0.500
#> 2      random       cauchy        0 1.000
#> 3 Sepal.Width       cauchy        0 0.123
#> 4 Petal.Width       cauchy        0 0.321
```

<sup>Created on 2020-09-17 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mattansb** [comment]
looks good
